aea/decision_maker/messages/transaction.py

```diff
@@ -77,6 +78,12 @@ def __init__(
         :param ledger_id: the ledger id
         :param info: a dictionary for arbitrary information
         """
+        skill_callback_ids = [
```

Comment:
a skill callback id can not only be a PublicId instance, but also a string that however must be convertible into a public id through the `PublicId.from_string`. 

That should improve the usability of this API. A similar feature has been implemented in `Envelope`. Probably it can be extended to other parts of the framework (so we can define a sort of `PublicIdLike` type).

I was thinking that it can be implemented roughly like this:
```python
class PublicId:

    def __new__(cls, *args, **kwargs):
        if len(args) == 1:
            return PublicId.from_string(args[0])
        if len(args) == 3:
            return PublicId(args[0], args[1], args[2])

```

Reply:
I am not sure we should add the `PublicIdLike` type....it leads to bloat and the benefits seem small to me.